<a href="https://colab.research.google.com/github/MedushaThiru/Internship---DecodeLabs/blob/main/Project1_Data_Cleaning_DecodeLabs.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Load and inspect the data


In [11]:
import pandas as pd

df = pd.read_excel('/content/Dataset for Data Analytics.xlsx')
print(df.shape)
print(df.head())
print(df.dtypes)
print(df.info())

(1200, 14)
     OrderID       Date CustomerID  Product  Quantity  UnitPrice  \
0  ORD200000 2023-01-04     C72649  Monitor         5     570.62   
1  ORD200001 2024-08-23     C75739    Phone         2     151.35   
2  ORD200002 2024-02-27     C81728   Tablet         5     550.68   
3  ORD200003 2023-10-15     C33540    Chair         1     273.19   
4  ORD200004 2025-05-08     C81840  Printer         4     626.01   

  ShippingAddress PaymentMethod OrderStatus TrackingNumber  ItemsInCart  \
0     928 Main St    Debit Card     Shipped    TRK37947903            7   
1     823 Main St        Online     Shipped    TRK91186779            3   
2     512 Main St   Credit Card   Cancelled    TRK42903982            8   
3     275 Main St    Debit Card    Returned    TRK62788070            5   
4     668 Main St        Online   Delivered    TRK29241424            8   

  CouponCode ReferralSource  TotalPrice  
0     SAVE10      Instagram     2853.10  
1     SAVE10       Referral      302.70  
2  

Identify missing/null values

In [12]:
print(df.isnull().sum())

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64


Decide how to handle the missing values

In [13]:
df['CouponCode'] = df['CouponCode'].fillna('No Coupon')
print(df['CouponCode'].value_counts())

CouponCode
FREESHIP     313
No Coupon    309
WINTER15     292
SAVE10       286
Name: count, dtype: int64


Check for duplicates

In [14]:
print("Full row duplicates:", df.duplicated().sum())
print("Duplicate OrderIDs:", df['OrderID'].duplicated().sum())

Full row duplicates: 0
Duplicate OrderIDs: 0


Fix data formats

In [15]:
# Date format
df['Date'] = pd.to_datetime(df['Date'])
print(df['Date'].min(), df['Date'].max())

# Text columns - check for inconsistent casing/whitespace
for col in ['Product', 'PaymentMethod', 'OrderStatus', 'ReferralSource', 'ShippingAddress']:
    print(col, df[col].unique()[:10])

# Strip whitespace and standardize case if needed
for col in ['Product', 'PaymentMethod', 'OrderStatus', 'ReferralSource']:
    df[col] = df[col].str.strip()

2023-01-01 00:00:00 2025-06-30 00:00:00
Product ['Monitor' 'Phone' 'Tablet' 'Chair' 'Printer' 'Laptop' 'Desk']
PaymentMethod ['Debit Card' 'Online' 'Credit Card' 'Gift Card' 'Cash']
OrderStatus ['Shipped' 'Cancelled' 'Returned' 'Delivered' 'Pending']
ReferralSource ['Instagram' 'Referral' 'Email' 'Facebook' 'Google']
ShippingAddress ['928 Main St' '823 Main St' '512 Main St' '275 Main St' '668 Main St'
 '934 Main St' '986 Main St' '706 Main St' '904 Main St' '102 Main St']


Validate numeric fields and round to 2 decimals

In [16]:
df['UnitPrice'] = df['UnitPrice'].round(2)
df['TotalPrice'] = df['TotalPrice'].round(2)

# Sanity check: does TotalPrice = Quantity * UnitPrice?
df['Calculated_Total'] = (df['Quantity'] * df['UnitPrice']).round(2)
mismatches = df[df['Calculated_Total'] != df['TotalPrice']]
print("Mismatches:", len(mismatches))

Mismatches: 0


Outlier scan (light pass, full analysis happens in Project 2)

In [17]:
print(df[['Quantity', 'UnitPrice', 'TotalPrice', 'ItemsInCart']].describe())

          Quantity    UnitPrice   TotalPrice  ItemsInCart
count  1200.000000  1200.000000  1200.000000  1200.000000
mean      2.945833   356.412750  1053.968300     5.485000
std       1.407557   197.177146   819.856558     2.281983
min       1.000000    11.390000    11.390000     1.000000
25%       2.000000   186.062500   410.520000     4.000000
50%       3.000000   364.210000   823.615000     5.000000
75%       4.000000   521.570000  1578.475000     7.000000
max       5.000000   699.930000  3456.400000    10.000000


Change Log table

| Change ID | Description | Impact | Status |
| :--- | :--- | :--- | :--- |
| CR001 | Filled 309 blank CouponCode values with "No Coupon" | 309 records | Resolved |
| CR002 | Verified 0 duplicate OrderIDs | 1200 records checked | Resolved |
| CR003 | Standardized Date to datetime/ISO format | All rows | Resolved |
| CR004 | Verified/rounded UnitPrice & TotalPrice to 2 decimals | All rows | Resolved |

Export the cleaned dataset

In [18]:
df.to_csv('Cleaned_Dataset_Project1.csv', index=False)
df.to_excel('Cleaned_Dataset_Project1.xlsx', index=False)

## ***This project addressed data quality issues in a 1,200-row e-commerce orders dataset before it could be used for analysis. Using Python (pandas), the dataset was checked for missing values, duplicate records, and formatting inconsistencies in dates and numeric fields. The audit found 309 missing values in the CouponCode column (resolved by labeling them "No Coupon"), zero duplicate Order IDs, and confirmed TotalPrice correctly matched Quantity × UnitPrice. The dataset is now clean, consistent, and ready for exploratory data analysis in Project 2.***